### PR TITLE
Paintroid-637 : Image marked as saved if saving did not work

### PR DESCRIPTION
**Pull request by vishad-tyagi**
Paintroid - 637 : Image marked as saved if saving did not work
https://catrobat.atlassian.net/browse/PAINTROID-637

Description

Paint something -> "save image" (on api > 28) -> Snackbar with errormessage appears -> "New image" -> image is deleted without dialog that asks to save

Upon attempting to save a painting, if the application lacks the requisite permissions to write to the device, a Snackbar notification correctly displays an error message indicating the issue. However, when the user opts to initiate a new image, the expected behaviour would be to present a dialog offering options to discard or save the current image. Contrary to this expectation, the application directly loads a new canvas without displaying the dialog, leading to the unintended loss of the user's unsaved work. 

## New Features and Enhancements
- No New Feature
## Refactorings and Bug Fixes
 Bug
- Previously workspace's last saved command count was updated after every save attempt, regardless of whether the save was successful.
For this, I Implemented : 
If the image is not successfully saved (!isFileSaved), it calls markUnsavedChanges to indicate there are pending changes in the workspace that haven't been successfully saved.
If the image is successfully saved, it updates the workspace's last saved command count using updateLastSavedCommandCount, similar to the original but only on a successful save.

#### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by vishad-tyagi** [comment]
@juliajulie95 There is an issue in one of the test cases. I am editing it. Rest all three PRs are working fine with all test cases passed. I apologise for this oversight. I will commit ASAP

**Comment by vishad-tyagi** [comment]
@juliajulie95 I have committed the changes. Good to go

**Comment by bakicelebi** [comment]
Hi @vishad-tyagi can you please update the PR description so that I can CR your ticket? Thanks.

**Review by msesko — APPROVED** [review]
Good work @vishad-tyagi ! I tested on both android and ios and the fix works well.

image

File: packages/features/workspace_screen/lib/src/states/workspace_state_notifier.dart [hunk 1]
```diff
@@ -20,10 +35,7 @@ class WorkspaceStateNotifier extends StateNotifier<WorkspaceState> {
     return result;
   }
 
-  void updateLastSavedCommandCount() => state = state.copyWith(
-        updatedLastSavedCommandCount: _commandManager.count,
-      );
-
+  // Toggle the fullscreen state
```
Comment (juliajulie95) [inline packages/features/workspace_screen/lib/src/states/workspace_state_notifier.dart]: comment should not be needed as the function name is descriptive enough

Comment (juliajulie95) [reply packages/features/workspace_screen/lib/src/states/workspace_state_notifier.dart]: Everything else looks good, please delete the comment then I can merge

Comment (vishad-tyagi) [reply packages/features/workspace_screen/lib/src/states/workspace_state_notifier.dart]: @juliajulie95 I have removed the comment

**Review by juliajulie95 — APPROVED** [review]
Good job, thank you!

**Pull request merged**